controllers/reconciler.go

```diff
@@ -446,6 +453,29 @@ func (r *SingleClusterReconciler) setStatusPhase(phase asdbv1.AerospikeClusterPh
 	return nil
 }
 
+func (r *SingleClusterReconciler) getClusterReadinessStatus() (bool, error) {
+	podList, err := r.getClusterPodList()
```

Comment:
> Are we assuming that this function will be called only when podList is not empty? If list is empty then it will return true.

Reply:
This func is called after all the racks are reconciled and the current readiness check is determined using all the existing pods